### PR TITLE
feat: made reset and identify api's synchronous to reflect the data immediately

### PR DESCRIPTION
**Pull request by desusai7**
# Made reset and identify api's synchronous to reflect the data immediately

### CODE REVIEW

**Review by pallabmaiti — DISMISSED** [review]

File: Tests/RSContextTests.swift [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+//  RudderTests
+//
+//  Created by Desu Sai Venkat on 13/03/23.
+//
+
+import Foundation
+import XCTest
+@testable import Rudder
+
+class RSContextTests: XCTestCase {
```
Comment (pallabmaiti) [inline Tests/RSContextTests.swift:12]: Get rid of RS prefix.

File: Tests/RSContextTests.swift [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        userSession.start()
+        context.setSessionData(userSession)
+        XCTAssertEqual(context.sessionId, userSession.getId())
+        XCTAssertTrue(context.sessionStart)
+    }
+    
+    func test_setConsentData() {
+        var deniedConsentIds = ["deniedId1", "deniedId2", "deniedId3", "deniedId4"]
+        context.setConsentData(deniedConsentIds)
+        XCTAssertEqual(context.getConsentData(), ["deniedConsentIds": deniedConsentIds])
```
Comment (pallabmaiti) [inline Tests/RSContextTests.swift:217]: Instead of getConsentData, persist the context object. You don't need this functions.

**Review by pallabmaiti — CHANGES_REQUESTED** [review]

File: Tests/RSContextTests.swift (hunk 1 above)
Comment (desusai7) [reply Tests/RSContextTests.swift]: Done

File: Tests/RSContextTests.swift [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+        deniedConsentIds = ["deniedId5", "deniedId6", "deniedId7", "deniedId8"]
+        context.setConsentData(deniedConsentIds)
+        XCTAssertEqual(context.getConsentData(), ["deniedConsentIds": deniedConsentIds])
+        deniedConsentIds = ["deniedId9", "deniedId10", "deniedId11", "deniedId12"]
+        context.setConsentData(deniedConsentIds)
+        XCTAssertEqual(context.getConsentData(), ["deniedConsentIds": deniedConsentIds])
+    }
+    
+    func convertToDictionary(text: String) -> [String: String]? {
```
[the remark this answers is not in the document]
Comment (desusai7) [reply Tests/RSContextTests.swift]: Moved the util methods

File: Tests/RSContextTests.swift (hunk 2 above)
Comment (desusai7) [reply Tests/RSContextTests.swift]: fixed this now

**Review by pallabmaiti — APPROVED** [review]

**Pull request merged**